_tests_/pet.test.js

```diff
@@ -0,0 +1,221 @@
+    expect(pet.hunger).toEqual(6);
+  });
+  it("decreases hunger to minimum value of 0 if hunger is less than 3", () => {
+    const pet = new Pet("Fido");
+    pet.hunger = 2;
+    pet.feed();
+    expect(pet.hunger).toEqual(0);
+  });
+});
+
+describe("checkUp", () => {
+  it("returns 'Your pet is no longer alive :(' if the pet is not alive", () => {
+    const pet = new Pet("Fido");
+    pet.fitness = 0;
+    pet.checkUp();
+    expect(pet.checkUp()).toEqual("Your pet is no longer alive :(");
+
+    pet.hunger = 10;
+    pet.checkUp();
+    expect(pet.checkUp()).toEqual("Your pet is no longer alive :(");
+
+    pet.age = 30;
+    pet.checkUp();
+    expect(pet.checkUp()).toEqual("Your pet is no longer alive :(");
+  });
+  it("returns 'I need a walk' when fitness is less than or equal to 3 and hunger is less than 5", () => {
+    const pet = new Pet("Fido");
+    pet.fitness = 3;
+    pet.hunger = 2;
+    pet.checkUp();
+    expect(pet.checkUp()).toEqual("I need a walk");
+  });
+  it("returns 'I am hungry' when hunger is greater than or equal to 5 and fitness is greater than 3", () => {
+    const pet = new Pet("Fido");
+    pet.fitness = 4;
+    pet.hunger = 5;
+    pet.checkUp();
+    expect(pet.checkUp()).toEqual("I am hungry")
+  });
+  it("returns 'I am hungry AND I need a walk' when fitness is less than or equal to 3 and hunger is greater than or equal to 5", () => {
+    const pet = new Pet("Fido");
+    pet.fitness = 3;
+    pet.hunger = 5;
+    pet.checkUp();
+    expect(pet.checkUp()).toEqual("I am hungry AND I need a walk")
+  });
+  it("returns 'I feel great' when fitness is greater than 3 and hunger is less than 5", () => {
+    const pet = new Pet("Fido");
+    pet.fitness = 4;
+    pet.hunger = 4;
+    pet.checkUp();
+    expect(pet.checkUp()).toEqual("I feel great!")
+  });
+});
+
+describe("adoptChild", () => {
+  it("throws an error if the pet is not alive", () => {
+    const pet = new Pet("Fido");
+    pet.fitness = 0;
+    expect(() => pet.feed()).toThrow("Your pet is no longer alive :(");
+            
+    pet.hunger = 10;
+    expect(() => pet.feed()).toThrow("Your pet is no longer alive :(");
+            
+    pet.age = 30;
+    expect(() => pet.feed()).toThrow("Your pet is no longer alive :(");
+  });
+  it("adds child object to parent object's children array", () => {
+    const pet = new Pet("Fido");
+    const baby = new Pet("Daisy");
+    pet.adoptChild(baby);
+    expect(pet.children).toEqual([{ name: "Daisy", age: 0, hunger: 0, fitness: 10, children: [] }]);
+  });
+});
+
+describe("haveBaby", () => {
+  it("throws an error if the pet is not alive", () => {
+    const pet = new Pet("Fido");
+    pet.fitness = 0;
+    expect(() => pet.feed()).toThrow("Your pet is no longer alive :(");
+            
+    pet.hunger = 10;
+    expect(() => pet.feed()).toThrow("Your pet is no longer alive :(");
+            
+    pet.age = 30;
+    expect(() => pet.feed()).toThrow("Your pet is no longer alive :(");
```

Comment:
as above, regardless of hunger or age the test is throwing the error due to `fitness = 0`